app/helpers/application_helper.rb

```diff
@@ -516,40 +516,30 @@ def category_badge(category, opts = nil)
     CategoryBadge.html_for(category, opts).html_safe
   end
 
-  def self.all_connectors
-    @all_connectors = Dir.glob("plugins/*/app/views/connectors/**/*.html.erb")
-  end
-
-  PLUGIN_OUTLET_TEMPLATE_CACHE = Concurrent::Map.new
+  SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX = Rails.env.test? ? "spec/fixtures/plugins/" : "plugins/"
+
+  SERVER_PLUGIN_OUTLET_CONNECTOR_TEMPLATES =
+    Dir
+      .glob("#{SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX}*/app/views/connectors/**/*.html.erb")
+      .reduce({}) do |connectors, template_path|
+        template_path =~ Regexp.new("/connectors/(.*)/.*\.html\.erb$")
+        outlet_name = Regexp.last_match(1)
+        connectors[outlet_name] ||= []
+        connectors[outlet_name] << template_path.sub(
+          "#{SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX}",
+          "",
+        ).delete_suffix(".html.erb")
+        connectors
+      end
 
   def server_plugin_outlet(name, locals: {})
     return "" if !GlobalSetting.load_plugins?
+    return "" if !SERVER_PLUGIN_OUTLET_CONNECTOR_TEMPLATES.key?(name)
 
-    matcher = Regexp.new("/connectors/#{name}/.*\.html\.erb$")
-    erbs = ApplicationHelper.all_connectors.select { |c| c =~ matcher }
-    return "" if erbs.blank?
-
-    erbs
-      .map do |erb|
-        cache_key = [erb, locals.keys.sort]
-
-        template =
-          PLUGIN_OUTLET_TEMPLATE_CACHE.compute_if_absent(cache_key) do
-            source = File.read(erb)
-            handler = ActionView::Template.handler_for_extension("erb")
-
-            ActionView::Template.new(
-              source,
-              "discourse_plugin_outlet__#{name}",
-              handler,
-              locals: locals.keys,
-              format: :html,
-              virtual_path: erb,
-            )
-          end
+    lookup_context.append_view_paths([Rails.root.join(SERVER_PLUGIN_OUTLET_PLUGINS_PREFIX)])
```

Comment:
This uses the following API: https://api.rubyonrails.org/classes/ActionView/LookupContext/ViewPaths.html#method-i-append_view_paths